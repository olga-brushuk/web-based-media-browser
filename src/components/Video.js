import React, { Component } from 'react';

class Video extends Component {

  render() {
    // depending on the source platform of the track generate different track id
    const trackLinkIdentifier = this.props.trackId;
    let trackId;

    if (trackLinkIdentifier.startsWith('/yt/')) { // Youtube source platform
      trackId = trackLinkIdentifier.replace('/yt/', ''); // get the id only

      return (
        <div className="track__video">
          <iframe type="text/html" width="640" height="360" frameBorder="no" title={this.props.trackName} 
          src={`https://www.youtube.com/embed/${trackId}`} allowFullScreen></iframe>
        </div>
      );
    } 
    else if (trackLinkIdentifier.startsWith('/sc/')) { // SoundCloud source platform
      // get the id only (maybe this can be improved or implemented in a different way, doesn't seem reliable)
      trackId = trackLinkIdentifier.split('tracks/')[1].slice(0, -7);
      
      return (
        <div className="track__video">
          <iframe width="100%" height="166" scrolling="no" frameBorder="no" title={this.props.trackName}
          src={`https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/${trackId}`}></iframe>
        </div>
      );
    } else { // fallback for other providers as it was not implemented yet
      return (
        <p>The track playback is not available</p>
      );
    }
  }
}

export default Video;