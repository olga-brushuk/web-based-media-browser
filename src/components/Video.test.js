import React from 'react';

import {configure, shallow} from 'enzyme';

import Video from './Video';

let wrapped = shallow(<Video trackId="/yt/U7KQ-8I_jYI" trackName="Brooks - If Only I Could" />);

describe('<Video />', () => {
    it('render correctly <Video /> component', () => {  
        expect(wrapped).toMatchSnapshot();
    });
    it('should render src of <Video /> item to be YouTube url with video id passed in trackId prop', () => {
        const iframe = wrapped.find('iframe');
        const props = iframe.props();
        expect(props.src).toEqual('https://www.youtube.com/embed/U7KQ-8I_jYI');
    });
});