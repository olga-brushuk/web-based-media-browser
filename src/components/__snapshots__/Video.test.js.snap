// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> render correctly <Video /> component 1`] = `
<div
  className="track__video"
>
  <iframe
    allowFullScreen={true}
    frameBorder="no"
    height="360"
    src="https://www.youtube.com/embed/U7KQ-8I_jYI"
    title="Brooks - If Only I Could"
    type="text/html"
    width="640"
  />
</div>
`;
